src/rgw/rgw_bucket.cc

```diff
@@ -2967,6 +2967,11 @@ int RGWMetadataHandlerPut_BucketInstance::put_check()
 
   bool exists = (!!orig_obj);
 
+  if (exists && make_private) {
+    /* don't sync bucket layout changes */
+    bci.info.layout = old_bci->info.layout;
+  }
```

Comment:
when `!exists` this would still take the remote zone's layout

since each zone is allowed to configure its own default shard count (whether by `rgw_override_bucket_index_max_shards` in ceph.conf, or by `radosgw-admin zone modify --bucket-index-max-shards`), i think we should use that to initialize the layout instead of taking the remote zone's default layout

Reply:
@cbodley fixed it. Thanks!